pkg/controllers/routercerts/controller.go

```diff
@@ -123,23 +145,13 @@ func (c *routerCertsDomainValidationController) validateRouterCertificates() ope
 		klog.Infof("system cert pool not available: %v", err)
 		verifyOptions.Roots = x509.NewCertPool()
 	}
-	for _, certificate := range certificates {
-		switch {
-		case certificate.IsCA && bytes.Equal(certificate.RawSubject, certificate.RawIssuer):
-			klog.V(4).Infof("using CA %s as root", certificate.Subject.String())
-			verifyOptions.Roots.AddCert(certificate)
-		case certificate.IsCA:
-			klog.V(4).Infof("using CA %s as intermediate", certificate.Subject.String())
-			verifyOptions.Intermediates.AddCert(certificate)
-		default:
-			serverCerts = append(serverCerts, certificate)
-		}
-	}
-
-	if len(verifyOptions.Roots.Subjects()) == 0 {
-		return newRouterCertsDegradedf("NoRootCARouterCerts", "secret/%v.spec.data[%v] -n %v: no root CA certificates found in secret or system", c.secretName, ingressDomain, c.targetNamespace)
+	// ignore the server cert from the default cert bundle
+	populateVerifyOptionsFromCertSlice(&verifyOptions, ingressCACerts)
```

Comment:
Even though the comment above states so, it might be easier to grep at first glance for future reviewers if you explicitly ignore the return value.
```suggestion
	_ = populateVerifyOptionsFromCertSlice(&verifyOptions, ingressCACerts)
```
Also (nit), consider renaming the method to better describe the full functionality:
```suggestion
	_ = populateVerifyOptionsAndGetServerCertsFromCertSlice(&verifyOptions, ingressCACerts)
```

Reply:
+1 for `_ = ...`

Can be a follow-up.